src/psyclone/psyir/nodes/omp_clauses.py

```diff
@@ -115,3 +117,223 @@ class OMPNogroupClause(Clause):
+
+        :return: whether the given child and position are valid for this node.
+        :rtype: bool
+
+        '''
+        return isinstance(child, Reference)
+
+
+class OMPFirstprivateClause(Clause):
+    '''
+    OpenMP firstprivate clause. This is used to declare variables as
+    firstprivate to an OpenMP region.
+    '''
+    _children_valid_format = "[Reference]*"
+    _text_name = "FirstprivateClause"
+
+    @property
+    def _clause_string(self):
+        if len(self.children) > 0:
+            return "firstprivate"
+        return ""
+
+    @staticmethod
+    def _validate_child(position, child):
+        '''
+        Decides whether a given child and position are valid for this node.
+        Any number of Reference nodes are allowed.
+
+        :param int position: the position to be validated.
+        :param child: a child to be validated.
+        :type child: :py:class:`psyclone.psyir.nodes.Node`
+
+        :return: whether the given child and position are valid for this node.
+        :rtype: bool
+
+        '''
+        return isinstance(child, Reference)
+
+
+class OMPDefaultClause(Clause):
+    '''
+    OpenMP Default clause. Used to determine the default declaration for
+    variables used in an OpenMP region.
+    '''
+
+    class DefaultClauseTypes(Enum):
+        '''Enumeration of the different types of OMPDefaultClause supported
+        in PSyclone'''
+        SHARED=0
+        NONE=1
+        FIRSTPRIVATE=2
+
+    _children_valid_format = None
+    _text_name = "OMPDefaultClause"
+
+    @property
+    def _clause_string(self):
+        clause_string = "default("
+        if self._clause_type == OMPDefaultClause.DefaultClauseTypes.SHARED:
+            clause_string = clause_string + "shared)"
+        elif self._clause_type == OMPDefaultClause.DefaultClauseTypes.NONE:
+            clause_string = clause_string + "none)"
+        elif (self._clause_type == 
+                OMPDefaultClause.DefaultClauseTypes.FIRSTPRIVATE):
+            clause_string = clause_string + "firstprivate)"
+        return clause_string
+
+
+    def __init__(self, clause_type=DefaultClauseTypes.SHARED):
+        if not isinstance(clause_type, OMPDefaultClause.DefaultClauseTypes):
+            raise TypeError(
+                    "OMPDefaultClause expected 'clause_type' argument of type "
+                    "OMPDefaultClause.DefaultClauseTypes but found '{0}'"
+                    .format(type(clause_type).__name__))
+        self._clause_type = clause_type
+        super(OMPDefaultClause, self).__init__()
+
+class OMPScheduleClause(Clause):
+    '''
+    OpenMP Schedule clause used for OMP Do Directives.
+    '''
+    _children_valid_format = "None"
+    _text_name = "OMPScheduleClause"
+    _schedule = ""
+
+    @property
+    def _clause_string(self):
+        return f"schedule({self._schedule})"
+
+    def set_schedule(self, schedule):
+        self._schedule = schedule
+
+    def __init__(self, schedule="static"):
```

Comment:
Should we have an enumeration of valid schedule values too or is that complicated?

Reply:
I was going to leave this for now since the existing code uses a string for the input, and there's no checking. I could add it but I think it should be a separate issue as this PR is already going to be large.